src/crimson/os/seastore/seastore.cc

```diff
@@ -127,16 +127,23 @@ using crimson::common::get_conf;
 SeaStore::Shard::Shard(
   std::string root,
   Device* dev,
-  bool is_test)
+  bool is_test,
+  unsigned int store_shard_nums,
+  unsigned int store_index)
   :root(root),
    max_object_size(
      get_conf<uint64_t>("seastore_default_max_object_size")),
    is_test(is_test),
    throttler(
-      get_conf<uint64_t>("seastore_max_concurrent_transactions"))
+      get_conf<uint64_t>("seastore_max_concurrent_transactions")),
+   store_index(store_index)
 {
-  device = &(dev->get_sharded_device());
-  register_metrics();
+  if(seastar::this_shard_id() + seastar::smp::count * store_index >= store_shard_nums) {
```

Comment:
"store_index is out of range {} - inactivating this store shard, store_shard_nums"